### ISSUE TITLE
Cannot run on example files

### ISSUE THREAD

**Issue by mctseng2** [owenjm/damidseq_pipeline#8]
Hello,

Thanks for offering this convenient package. This is our first attempt to analyze dam-id samples. After installing, I tried to run on the example files you provided but it couldn't run:

`damidseq_pipeline     --gatc_frag_file=/home/groups/hpcbio/RNA-Seq/projects/lixin/2019JunDam-ID-Seq/data/genome/gatc_file/     --bowtie2_genome_dir=/home/groups/hpcbio/RNA-Seq/projects/lixin/2019JunDam-ID-Seq/data/genome/bowtie2-dmel-r6.28/dmel-r6.28     --threads=1`

Following is the output I got:

`*** Reading data files ***

Index   Name
A5      polII
A12     Dam

*** Matching adaptors ***
SLX-7564.C2FUMACXX.s_3.r_1.Index12.sequence.cut.fastq.gz
Dam: Index 12
SLX-7564.C2FUMACXX.s_3.r_1.Index5.sequence.cut.fastq.gz
polII: Index 5

*** Aligning files with bowtie2 ***
Now working on Dam ...
3000 reads; of these:
  3000 (100.00%) were unpaired; of these:
    613 (20.43%) aligned 0 times
    2115 (70.50%) aligned exactly 1 time
    272 (9.07%) aligned >1 times
79.57% overall alignment rate

Now working on polII ...
3000 reads; of these:
  3000 (100.00%) were unpaired; of these:
    847 (28.23%) aligned 0 times
    1858 (61.93%) aligned exactly 1 time
    295 (9.83%) aligned >1 times
71.77% overall alignment rate

*** Reading GATC file ***
  Sorting ...

*** Extending reads up to 300 bases ***
Reading input file: Dam ...
  Processing data ...
  Warning: alignment contains chromosome identities not found in GATC file (see log file for details; this is normal if unmapped assemblies and the mitochondrial genome have been excluded from the GATC file.)
  Seqs extended (>q30) =

Reading input file: polII ...
  Processing data ...
  Warning: alignment contains chromosome identities not found in GATC file (see log file for details; this is normal if unmapped assemblies and the mitochondrial genome have been excluded from the GATC file.)
  Seqs extended (>q30) =

*** Calculating bins ***
Now working on Dam-ext300 ...
Generating .bam file ...
  sorting ...
[bam_sort] Use -T PREFIX / -o FILE to specify temporary and final output files
Usage: samtools sort [options...] [in.bam]
Options:
  -l INT     Set compression level, from 0 (uncompressed) to 9 (best)
  -m INT     Set maximum memory per thread; suffix K/M/G recognized [768M]
  -n         Sort by read name
  -o FILE    Write final output to FILE rather than standard output
  -T PREFIX  Write temporary files to PREFIX.nnnn.bam
  -@, --threads INT
             Set number of sorting and compression threads [1]
      --input-fmt-option OPT[=VAL]
               Specify a single input file format option in the form
               of OPTION or OPTION=VALUE
  -O, --output-fmt FORMAT[,OPT[=VAL]]...
               Specify output format (SAM, BAM, CRAM)
      --output-fmt-option OPT[=VAL]
               Specify a single output file format option in the form
               of OPTION or OPTION=VALUE
      --reference FILE
               Reference sequence FASTA FILE [null]
[E::hts_open_format] fail to open file 'Dam-ext300.bam'
samtools view: failed to open "Dam-ext300.bam" for reading: No such file or directory
Error: no reads obtained from bamfile ... exiting.
`

Looks like the alignment was successful but it failed to manage sam files. Do you have any clue for this issue? Thanks a lot!

Meng-Chun

**Comment by mctseng2** [comment]
I have figured out some clue. For some reason, the pipeline thinks I have samtools 1.2 loaded while I was using 1.7. Obviously, the calls for samtools 1.2 and >1.2 are different. The pipeline seems to execute 1.2 commands on 1.7. I re-loaded with samtools 1.2 and problems were solved. However, I did not know why the pipeline thinks I have samtools 1.2 loaded.

**Comment by owenjm** [comment]
That's very odd.  Yes, thanks to the bizarre change in options format in
samtools post v1.2, we have to check for the version and adjust
accordingly.  The log file should list the version of samtools it found;
the default if no version information is detected is assumed to be 1.2 (but
you should see a warning on the command-line if this is the case -- I
thought I had sanity checks in there to catch all options!)

Is there any chance you might have two versions of samtools installed on
your system?  The log file should also list which version of samtools was
detected -- can you confirm that the version was detected correctly?

Cheers,
Owen

**Comment by mctseng2** [comment]
Thanks for your response. I checked the logfile and it did not say anything (it looks identical with the output message I posted above) about the version of samtools. I guess it did not detect anything so it assumed version 1.2 (I was running the pipeline under the HPC cluster settings where the software modules need to be loaded before executing them. )? I think it would be great if the samtools version can be specified as a command option.
